calicoctl/calicoctl/commands/node/diags.go

```diff
@@ -229,6 +229,15 @@ func writeDiags(cmds diagCmd, dir string) {
 
 	content, err := exec.Command(parts[0], parts[1:]...).CombinedOutput()
 	if err != nil {
+		// sometimes socket information is not collected as netstat tool
```

Comment:
Im not a huge fan of putting command-specific code in here - that could get messy quick. 

I think we have two options:
- Return an error when the command fails, and have the calling code handle it (either ignore, or try a different command)
- Always run both `netstat` and `ss` so that we get at least one of them in the diags bundle (even though one or the other might fail). 

The first option is certainly the more "correct" solution, but the second is easier to implement. What do you think?

Reply:
thanks! I prefer the first option.

Reply:
but if we use first option, we need to handle it at the caller. for example, when the `ss` command doesn't fails, we should call the `ioutil.WriteFile()` on the caller, just like to:https://github.com/projectcalico/calico/blob/0e21c65dc326defce3f0ad089182d0007f415658/calicoctl/calicoctl/commands/node/diags.go#L251

so does it look less elegant ?